react/features/token-data/actions.js

```diff
@@ -29,8 +29,14 @@ export function parseTokenData() {
         // the iss field is not anonymous. If you want to pass data with
         // JWT token but you don't want to pass the JWT token for
         // verification the iss field should be set to "anonymous".
-        if (!config.token && payload && payload.iss !== 'anonymous') {
-            config.token = token;
+        if (payload && payload.iss !== 'anonymous') {
+            parsedToken.token = token;
```

Comment:
I'd say we always keep the raw around, and whomever cares must check the issuer before passing it along, the responsibility of the TokenData class is to parse the token, nothing else.